haskell/assets/ghci_script

```diff
@@ -1 +1,29 @@
 :add {ADD_SOURCES}
+:module + System.IO GHC.IO.Handle Control.Exception System.Directory
+rules_haskell_stdout_dupe <- hDuplicate stdout
+:{
+(rules_haskell_stdout_copy_file, rules_haskell_stdout_copy_h) <- do
+  (fn, h) <- openTempFile "/tmp" "rules-haskell-ghci-repl"
+  hDuplicateTo h stdout
+  return (fn, h)
+:}
+:show modules
+:{
+rules_haskell_loaded_modules <- do
+  hClose rules_haskell_stdout_copy_h
+  h <- openFile rules_haskell_stdout_copy_file ReadMode
+  xs <- hGetContents h
+  removeFile rules_haskell_stdout_copy_file
+  return $ takeWhile (/= ' ') <$> lines xs
+:}
+hDuplicateTo rules_haskell_stdout_dupe stdout
+:{
+let rules_haskell_add_loaded_modules _ =
+      return $ ":module + " ++
+        Data.List.intercalate " " (("*" ++) <$> rules_haskell_loaded_modules)
+:}
+:module - System.IO GHC.IO.Handle Control.Exception System.Directory
+:def rules_haskell_add_loaded_modules rules_haskell_add_loaded_modules
+:rules_haskell_add_loaded_modules
+:undef rules_haskell_add_loaded_modules
+:reload
```

Comment:
Maybe this `:reload` should be removed, I'm not sure if it leads to increased startup times for REPLs or not.

Reply:
What's its purpose?

Reply:
This reloads modules and as a side-effect removes all the temporarily defined `rules_haskell_*` functions.

Reply:
In my experience this is pretty quick, if the modules haven't changed.